docs/device_support_guide.md

```diff
@@ -81,21 +79,83 @@ If you are lucky, your laptop model will have similar addresses to another lapto
 that is already supported by the driver, but usually this only happens on some
 features like cooler boost and battery charge limit.
 
+On Linux, you can do the EC dump in several ways.
++ `msi-ec` built-in debug mode (git version only)
++ `ec-sys` kernel module
++ By reading the memory that is virtually mapped to EC
+
+To request support for your device model, make the EC dump in text form
+and open the [Issue on Github](https://github.com/BeardOverflow/msi-ec/issues/new?assignees=&labels=New+firmware&projects=&template=support_request.yml).
+
+### `MSI-EC` debug mode
+
+Make sure that you are running the git version of the module.
+If not, follow the installation guide in the [Readme](../README.md#Installation) file.
+
+Unload module if it was previously loaded:
+
++ `sudo modprobe -r msi-ec`
+
+Load module with debug mode enabled:
+
++ `sudo modprobe msi-ec debug=true`
+
+Make a dump by reading the module pseudo file:
+
++ `cat /sys/devices/platform/msi-ec/debug/ec_dump`
+
+Or save it to file:
+
++ `cat /sys/devices/platform/msi-ec/debug/ec_dump > ec_dump.txt`
+
+If you got the next error, then your module is not loaded in debug mode or doesn't have it:
+
+> `cat: /sys/devices/platform/msi-ec/debug/ec_dump: No such file or directory`
+
+### `EC-SYS` method
+
+> [!NOTE]
+> The `ec-sys` module may not be included in some distros, such as Fedora.
+
 To start, You need to load a module called `ec_sys`:
 
 * `sudo modprobe ec_sys`
 
-if you need to write to a specific address (but you really shouldn’t) you can enable
-Read/Write mode for this module:
+[//]: # (If you need to write to a specific address &#40;but you really shouldn’t&#41; you can enable)
+[//]: # (Read/Write mode for this module:)
+[//]: # ()
+[//]: # (* `sudo modprobe ec_sys write_support=1`)
 
-* `sudo modprobe ec_sys write_support=1`
-
-After that you can extract the EC table and print it on the terminal:
+After that you can extract the EC table and print it on the terminal in text form:
 
 * `hexdump -C /sys/kernel/debug/ec/ec0/io`
 
-or you can put it in a .txt file in your home folder directly:
+Or save it to file:
+
++ `hexdump -C /sys/kernel/debug/ec/ec0/io > ec_dump.txt`
+
+If you need dump in binary form:
+
+* `cat /sys/kernel/debug/ec/ec0/io > ec_dump.bin`
+
+### Reading the EC RAM mapped to system memory
+
+Some devices have EC memory mapped to system memory address `0x0xFC000800`,
+so you can read it from the `/dev/mem` pseudo file with `dd`.
+
+`/dev/mem` may not be supported on some distros. For details, see `man mem`.
+
+> [!WARNING]
+> Reading random parts of the system memory can reveal your secrets, so check the dump before you post it to Github.
```

Comment:
and how do i check if the dump i made doesn't include sensitive info? let's call this method "experimental" until we make sure it works no less than 90% of the time.

Reply:
true, only by comparing with reference dump, like matching the location of the ec name

Reply:
Actually, chance of getting reading error is higher than accidentally dump some confidential info. And maybe this memory area newer mapped to system RAM

Reply:
Let's mark it as "Not recommended" and put the warning at the top of this method's section, so it can be the last resort if the two previous methods didn't work for some magical reason